Fotoblog.BLL/Services/PhotosService/PhotoService.cs

```diff
@@ -0,0 +1,185 @@
+﻿using AutoMapper;
+using Fotoblog.BLL.Services.ServiceResultNS;
+using Fotoblog.DAL;
+using Fotoblog.DAL.Entities;
+using Fotoblog.Utils.ViewModels.Photos;
+using Microsoft.AspNetCore.Http;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.Processing;
+using System.IO;
+using System.Net;
+
+namespace Fotoblog.BLL.Services.PhotosService
+{
+    public class PhotoService : BaseService, IPhotoService
+    {
+        private readonly string[] _allowedExtensions = { ".jpg", "jpeg", ".png" };
+        private readonly string[] _allowedImgContentTypes = { "image/jpeg", "image/png" };
+        private string? _uploadedFileExtension;
+        private string _saveLocation = "c:\\uploads"; // random folder added later
+        private readonly int _thumbnailWidth = 600;
+        public PhotoService(ApplicationDbContext dbContext, IMapper mapper) : base(dbContext, mapper)
+        {
+        }
+
+        public async Task<ServiceResult> AddPhoto(NewPhotoDataVm newPhotoDataVm)
+        {
+            try
+            {
+                doInputValidation(newPhotoDataVm);
+                PreparePhotoSaveLocation();
+            }
+            catch(Exception)
+            {
+                return ServiceResult.Fail(ErrorCodes.CouldNotSaveTheFile);
+            }
+                
+
+            var origSavePath = Path.Combine(_saveLocation, $"orginal{_uploadedFileExtension}");
+            var thumbnailSavePath = Path.Combine(_saveLocation, $"thumbnail{_uploadedFileExtension}");
+
+            try
+            {
+                await saveOriginalPhoto(origSavePath, newPhotoDataVm.file);
+                await saveThumbnail(thumbnailSavePath, newPhotoDataVm.file);
+            }
+            catch (Exception)
+            {
+                return ServiceResult.Fail(ErrorCodes.CouldNotSaveTheFile);
+            }
+
+            await SaveEntity(newPhotoDataVm);
+
+            return ServiceResult.Ok();
+        }
+
+        private async Task SaveEntity(NewPhotoDataVm newPhotoDataVm)
+        {
+            var entity = new PhotoEntity();
+            entity.Title = newPhotoDataVm.title;
+            entity.Description = newPhotoDataVm.description;
+            entity.ImagePath = _saveLocation;
+            entity.CreateDate = DateTime.Now;
```

Comment:
CreateDate może być uzupełniany automatycznie przez context przy zapisie

Reply:
pamiętam że tak robiliśmy, ale to w BaseEntity był ten property, natomiast w tym projekcie nie każde entity będzie go mieć. Chyba że masz coś innego na myśli, to możemy o tym pogadać jeszcze

Reply:
Każde entity może mieć. Możesz z tego pola nie korzystać w pewnych przypadkach. Lepiej tak niż za każdym razem musieć pamiętać u wypełnieniu daty.

Reply:
co racja to racja :) done